### PR TITLE
[log] Migrate `go/vt/log` flags to `pflag`

### LINKED ISSUE #10808: Switch flag definitions to be on pflag instead of flag in `package go/vt/log`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/log/*.go
        flag.Uint64Var(&glog.MaxSize, "log_rotate_max_size", glog.MaxSize, "size in bytes at which logs are rotated (glog.MaxSize)")
```

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This migrates the flags defined in `go/vt/log` to pflag. Similar to #11028, the flag registration has to happen backwards, and then for double-fun, (a) every binary needs logging and (b) not every binary uses `servenv.ParseFlags(WithArgs)?`, so this one needed some additional special handling, on top of the aforementioned special handling.

## Related Issue(s)

Closes #10808.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/log/log.go [hunk 1]
```diff
@@ -73,6 +72,11 @@ var (
 	FatalDepth = glog.FatalDepth
 )
 
-func init() {
-	flag.Uint64Var(&glog.MaxSize, "log_rotate_max_size", glog.MaxSize, "size in bytes at which logs are rotated (glog.MaxSize)")
+// RegisterFlags installs log flags on the given FlagSet.
+//
+// `go/cmd/*` entrypoints should either use servenv.ParseFlags(WithArgs)? which
```
Comment (deepthi) [inline go/vt/log/log.go:77]: Why the `?`

File: go/cmd/vtorc/main.go [hunk 2]
```diff
@@ -101,6 +102,7 @@ func main() {
 	// TODO(ajm188): after v15, remove this pflag hack and use servenv.ParseFlags
 	// directly.
 	fs := pflag.NewFlagSet("vtorc", pflag.ExitOnError)
+	vtlog.RegisterFlags(fs)
```
Comment (deepthi) [inline go/cmd/vtorc/main.go:105]: @GuptaManan100 @rsajwani IIUC this should go away when we change vtorc to start using servenv.

**Review by deepthi — APPROVED** [review]
Wow, we have a lot of binaries that don't use servenv.
I had to review this one commit at a time, but LGTM.

File: go/cmd/vtorc/main.go (hunk 2 above)
Comment (GuptaManan100) [reply go/cmd/vtorc/main.go:105]: I would think yes, since serenv already has these flags registered

File: go/vt/log/log.go (hunk 1 above)
Comment (ajm188) [reply go/vt/log/log.go:77]: regex: ParseFlags + ParseFlagsWithArgs

File: go/cmd/vtorc/main.go (hunk 2 above)
Comment (ajm188) [reply go/cmd/vtorc/main.go:105]: yes, anyone that uses `servenv` to parse flags (WithArgs or not) doesn't need to do this manually

**Review by GuptaManan100 — APPROVED** [review]

**Review by rsajwani — COMMENTED** [review]
LGTM

**Pull request merged**